.Rprofile

```diff
@@ -1,45 +1,22 @@
-# newest libraries anymore). By default it is not active.
-
-# Gather all library snapshots
-snapshots <- sort(list.files(
-             path = '/p/projects/rd3mod/R/libraries/snapshots',
-             pattern = '^20[0-9]{2}_[0-9]{2}(_[0-9]{2})?$',
-             full.names = TRUE), decreasing = TRUE)
-latest_snapshot <- ifelse (length(snapshots) > 0, snapshots[[1]], NA)
-
-# Just uncomment the following line and set the snapshot path
-# to a path of your choice or use the second line to use the latest snapshot.
-# Please make also sure that in your config file this .Rprofile file is copied
-# to the model output folder. Otherwise, the run itself will again use the
-# default library set!
-
-# snapshot <- "/p/projects/rd3mod/R/libraries/snapshots/2022_03"
-# snapshot <- latest_snapshot
-
-if(exists("snapshot") && file.exists(snapshot)) {
-  message("libPaths was manually set to: ",snapshot)
-  .libPaths(snapshot)
-}
-
-# Check if the library folder is currently being updated and if so use latest daily or monthly snapshot.
-if (any(grepl("^00LOCK.*", list.files(.libPaths()[1])))) {
-    lock_folders <- grep("^00LOCK.*", list.files(.libPaths()[1]), value=TRUE)
-    bad_packages <- gsub("^00LOCK-","", lock_folders)
-
-    # Give user diagnosis
-    message("\nThe following lock folders were found at ", .libPaths()[1], ":\n  ", lock_folders)
-    message("That means that the ", bad_packages, "package(s) is(are) currently being updated.")
-    message("All packages will be loaded from the library's latest snapshot instead:\n  ", latest_snapshot)
-    message("(If the lock folder isn't deleted automatically in the next couple of minutes, that means the package failed to update/install and that the folder has to be removed manually!)")
-    message("To avoid this automatic choice, specify a snapshot in your .Rprofile instead.")
-
-    if(file.exists(latest_snapshot)) {
-        .libPaths(latest_snapshot)
-    }
+source("renv/activate.R")
+
+# TODO determine pikPiamPackages dynamically
+pikPiamPackages <- c("GDPuc", "HARr", "MagpieNCGains", "citation", "demystas", "edgeTransport",
+                    "edgeTrpLib", "gdx", "gdxrrw", "gms", "goxygen", "iamc", "limes",
+                    "lpjclass", "lucode2", "luplot", "luscale", "lusweave", "madrat",
+                    "magclass", "magpie4", "magpiesets", "mip", "modelstats", "mrcommons",
+                    "mrdrivers", "mredgebuildings", "mrfable", "mrfeed", "mrfish",
+                    "mrland", "mrmagpie", "mrremind", "mrsoil", "mrtutorial", "mrvalidation",
+                    "mrvalidnitrogen", "mrwaste", "mrwater", "mstools", "piamModelTests",
+                    "piktests", "quitte", "regressionworlddata", "remind", "remind2",
+                    "remulator", "rmndt", "shinyresults", "trafficlight")
+
+
+# TODO should this be on or off by default?
+# comment out the following line to disable auto-updates
+renv::update(intersect(utils::installed.packages()[, "Package"], pikPiamPackages))
```

Comment:
This line defeats the whole purpose of the commit.

Reply:
I'm quite sure this is not problematic unless multiple R sessions are running in parallel. Updating package x only breaks R when the update is installed while x is already loaded. When updating at the start of the session via .Rprofile no packages are loaded yet.

Reply:
No.  I did not write the line does not work.  

> This PR replaces the snapshot machinery in .Rprofile with an renv based approach.

If the `renv` library gets updated every time R is started in this particular directory, that is the exact opposite of what the "snapshot machinery" should provide – stable package versions.